lib/pbench/server/api/resources/upload_api.py

```diff
@@ -22,29 +27,174 @@
 
 
 class HostInfo(Resource):
+    """
+    Show the status of the server.
+
+    This code will read a standard file and return a stock 200 / OK status if
+    the content doesn't start with `MESSAGE==`; and, if it does, fail with
+    503 / Service Unavailable and the remaining text of the file as a
+    `"message"` response key.
+
+    TODO: This is an artifact of the old ssh-based mechanism, where a fixed
+    symlink on disk points either to a templated file with the scp target
+    directory or a status string prefixed with `MESSAGE===` intended to be
+    reported to the user as an explanation that the server is not currently
+    taking calls -- e.g. (in standard boilerplate) because it's "in
+    maintenance mode". We should get away from the fixed files, and we have no
+    need for the templated scp directory as we no longer support ssh uploads.
+    We should create a new ADMIN-only server status API to specify whether the
+    server should be active or not, with an optional message that will be
+    reported by this API. This could become part of a "server state"
+    PostgreSQL table.
```

Comment:
An incremental step that leverages the existing install/ansible setup in a slightly less archaic manner (since the SSH upload path is meaningless now). I think I created an issue to work out a mechanism to save/manage this sort of internal server state; though I can't find it right now.

Reply:
OK...if you think this is an incremental step towards where we want to be, then I want to believe you.  (To me, it looks like a step back towards what we used to have which we no longer want....)